src/fsharp/FSComp.txt

```diff
@@ -1372,6 +1372,7 @@ keywordDescriptionLazy,"Used to specify a computation that is to be performed on
 keywordDescriptionLet,"Used to associate, or bind, a name to a value or function."
 keywordDescriptionLetBang,"Used in asynchronous workflows to bind a name to the result of an asynchronous computation, or, in other computation expressions, used to bind a name to a result, which is of the computation type."
 keywordDescriptionMatch,"Used to branch by comparing a value to a pattern."
+keywordDescritpionMatchBang,"Used in asynchronous workflows (and other computation expressions) to first bind the result of the asynchronous workflow (or computation expression), then pattern match over that result"
```

Comment:
I think it might be better to not mention async CEs here (even if they're likely to be the most common use case). Perhaps something along these lines?

"Used in a computation expression to pattern match directly over the result of a call to another computation expression."

Reply:
I was doing that to be consistent with the descriptions for the other "bang" keywords; they all seem to mention asyncs. Would this be an issue?

Reply:
Newbies will be scared by the repeated use of the term computation expression. Also, that still doesn't clearly explain what match! does IMHO.  How about just "Shorthand for let! and then match"?

Reply:
Newbies will be scared by the repeated use of the term computation expression. Also, that still doesn't clearly explain what match! does IMHO.  How about just "Shorthand for let! and then match"?

Reply:
@jwosty It's actually all inconsistent 😢 - `let!` mentions async, but `yield!` is just about CEs. I think the keyword descriptions here should be generic in its description, just like the feature itself.

@isaacabraham Have you experienced that with messages like what we have in `yield!`?

"Used in a computation expression to append the result of a given computation expression to a collection of results for the containing computation expression."

How would you suggest changing this to only mention CEs once?

Reply:
@cartermp OK, as long as we're being consistently inconsistent. :silly: (kidding, maybe we can fix the others some time). I personally would prefer them not to mention async's as well; I'll make that change.

In any case I think I like your suggestion.

Reply:
Maybe a merging of those two ideas? e.g. something like

"Used in a computation expression to pattern match directly over the result of a call to another computation expression. Equivalent to let! followed by match"